### PR TITLE
stix2matcher.matcher usage optimization

### PR DESCRIPTION
**Pull request by baulus**

### CODE REVIEW

File: stix_shifter_modules/error_test/stix_transmission/connector.py [hunk 1]
```diff
@@ -33,19 +32,9 @@ def __init__(self, connection, configuration):
...
-        if found_bindings:
-            matching_sdos = []
-            for binding in found_bindings:
-                matches = [match for match in matcher.get_sdos_from_binding(binding) if match not in matching_sdos]
-                matching_sdos.extend(matches)
-        else:
-            matching_sdos = []
-
+        compiled_pattern = Pattern(pattern)
+        matching_sdos = compiled_pattern.match(observed_data_sdos, verbose)
```
Comment (mdazam1942) [inline stix_shifter_modules/error_test/stix_transmission/connector.py:36]: so the match() doesn't return just the first anymore?

File: stix_shifter_modules/stix_bundle/stix_transmission/connector.py [hunk 2]
```diff
@@ -31,19 +30,7 @@ def __init__(self, connection, configuration):
...
-
-        found_bindings = matcher.matched()
-
-        if found_bindings:
-            matching_sdos = []
-            for binding in found_bindings:
-                matches = [match for match in matcher.get_sdos_from_binding(binding) if match not in matching_sdos]
-                matching_sdos.extend(matches)
-        else:
-            matching_sdos = []
```
Comment (mdazam1942) [inline stix_shifter_modules/stix_bundle/stix_transmission/connector.py:45]: so the match() doesn't return just the first anymore?

Comment (mdazam1942) [reply stix_shifter_modules/stix_bundle/stix_transmission/connector.py:45]: looks like this optimization introduced the original problem where the match() function doesn't return all the matching observables.

Comment (baulus) [reply stix_shifter_modules/stix_bundle/stix_transmission/connector.py:45]: there was such problem before?

**Pull request closed without merging**